### ISSUE TITLE
Digits in sequence names breaking NG86 parser in Bio.Phylo.PAML

### ISSUE THREAD

**Issue by Zefeng-Wu** [biopython/biopython#1290]
File "/usr/local/lib/python2.7/dist-packages/Bio/Phylo/PAML/_parse_yn00.py", line 44, in parse_ng86
  NG86["dN"] = line_floats[i + 1] 

I think the 22th line of _parse_yn00.py  has a  bug,  because there are digits in my sequences names. The script mistake the sequence name as lines_floats lines.

**Comment by peterjc** [comment]
Which version of Biopython are you using?

```
>>> import Bio
>>> print(Bio.__version__)
...
```

Can you share a sample file which we could us for a test case?

CC @brandoninvergo

**Comment by chmccarthy** [comment]
Hi,

I've encountered the same problem as the OP and I think it might due to the presence of periods in sequence names rather than digits or other symbols. I'm using yn00 to analyze FASTA-formatted CDS alignments which have sequence headers in the format Genome|ContigID_Start_Stop (e.g. "H222|CP028442.1_6509_9127", [I've attached an example file](https://github.com/biopython/biopython/files/2877157/test_fasta.txt)). These are run through yn00 using the following code I'm currently testing out:

>     yn = yn00.Yn00(alignment="test_fasta.txt", out_file="test.yn00")
>     yn.set_options(verbose=0, icode=0, weighting=0, commonf3x4=0)
>     results = yn.run(ctl_file=None, command="yn00", parse=True)
>     print results

When periods are included in the sequence headers I get the same error as the OP, but when I remove them yn00 runs and its output is parsed without any issue. I think that the periods are throwing off the regex parser in the parse_ng86 function in _parse_yn00.py? My nucleotide alignments ultimately derived from GenBank data, and all sequences from there follow the [ACCESSION.VERSION](https://www.ncbi.nlm.nih.gov/genbank/samplerecord/#VersionB) format for sequence IDs so that might be what's causing the issue.

For reference, I'm using Biopython 1.73 and Python 2.7.10.

**Comment by peterjc** [comment]
Your description does sound like dots are the problem - good detective work @chmccarthy 

My guess is that the regular expression to match a data value is wrongly picking up on the numbers dot numbers in your sequence name:

https://github.com/biopython/biopython/blob/biopython-173/Bio/Phylo/PAML/_parse_yn00.py#L22

@brandoninvergo could you look at this, or perhaps advise on how it might best be fixed? Thanks!

**Comment by brandoninvergo** [comment]
Egads, I completely missed the notification for this. Many apologies. @Zefeng-Wu and @chmccarthy, thanks for reporting this and digging into it a bit.

I built some test cases and it only seems to be triggered on names that contain bits that can be misinterpreted as floating point values by the parser (so, @chmccarthy, in your example, H222|CP028442.1_6509_9127 looks suspiciously like 28442.1 to the parser), which then get added to the list of floating point values on the line. The solution is just to pick the float values out of the rest of the line after we've already grabbed the sequence ID.

Pull request incoming.
